src/overlays/actors/ovl_En_Ta/z_en_ta.c

```diff
@@ -788,148 +866,153 @@ void func_80B1585C(EnTa* this, PlayState* play) {
         }
     }
 
-    if (this->unk_2CC == 0) {
-        func_80B13AA0(this, func_80B154FC, func_80B16938);
-        this->unk_2E0 &= ~0x10;
+    if (this->timer == 0) {
+        EnTa_SetupAction(this, EnTa_RunCuccoGame, EnTa_AnimRunToEnd);
+        this->stateFlags &= ~TALON_STATE_FLAG_ANIMATION_FINISHED;
         Animation_Change(&this->skelAnime, &gTalonSitWakeUpAnim, 1.0f,
                          Animation_GetLastFrame(&gTalonSitWakeUpAnim) - 1.0f,
                          Animation_GetLastFrame(&gTalonSitWakeUpAnim), ANIMMODE_ONCE, 10.0f);
         func_8002DF54(play, &this->actor, 7);
     }
 }
 
-void func_80B15AD4(EnTa* this, PlayState* play) {
-    if (this->unk_2CC == 0 && this->unk_2E0 & 0x20) {
-        func_80B13AA0(this, func_80B1585C, func_80B16938);
-        this->unk_2E0 &= ~0x10;
+void EnTa_StartingCuccoGame3(EnTa* this, PlayState* play) {
+    if (this->timer == 0 && this->stateFlags & TALON_STATE_FLAG_CUCCO_GAME_START_EVENT_TRIGGERED) {
+        EnTa_SetupAction(this, EnTa_ThrowSuperCuccos, EnTa_AnimRunToEnd);
+        this->stateFlags &= ~TALON_STATE_FLAG_ANIMATION_FINISHED;
+        // Play hand raise animation again so that it looks like Talon throws the cuccos
         Animation_Change(&this->skelAnime, &gTalonSitHandsUpAnim, 1.0f, 1.0f,
                          Animation_GetLastFrame(&gTalonSitHandsUpAnim), ANIMMODE_ONCE, 0.0f);
-        this->unk_2CC = 50;
-        func_80088B34(0x1E);
-        func_800F5ACC(NA_BGM_TIMED_MINI_GAME);
-        this->unk_2E0 |= 0x200;
+        this->timer = 50;
+
+        func_80088B34(30);                     // Set the minigame timer for 30 seconds
+        func_800F5ACC(NA_BGM_TIMED_MINI_GAME); // Start minigame music
```

Comment:
imo no need to comment on these, only cause it will just be something to remove later when the functions are properly named. I agree it helps with understanding short term, but thinking longer term this will be self explanatory